From 3646fc2ddcf033a85f0cfd61d3f09ca696b63023 Mon Sep 17 00:00:00 2001
From: Tamal <tasaha@habana.ai>
Date: Sat, 10 Aug 2024 16:16:52 +0300
Subject: [PATCH] [SW-161727] g3: remove delay post ist read, write

This patch removes delay after ist write and
read. This will improve bist execution time.

Change-Id: I51cb3294f3d09e6ec6a44993780a5c2f5cf5a181
Signed-off-by: Tamal <tasaha@habana.ai>
---

diff --git a/arch/gaudi3/arc/asic.c b/arch/gaudi3/arc/asic.c
index 0a37c41..6140eeb 100644
--- a/arch/gaudi3/arc/asic.c
+++ b/arch/gaudi3/arc/asic.c
@@ -53,7 +53,6 @@
 		preboot_puts("Die 0 cache memories MBIST fail\n");
 	} else {
 		bist_data.bist_result = SRAM_MBIST_ALL_PASS;
-		/* TODO: update "skip" by "pass" after memcpy engine timeout resolved */
 		preboot_puts("Die 0 cache memories MBIST Pass\n");
 	}
 }
diff --git a/arch/gaudi3/arc/mbist.c b/arch/gaudi3/arc/mbist.c
index 8570534..04d4ec8 100644
--- a/arch/gaudi3/arc/mbist.c
+++ b/arch/gaudi3/arc/mbist.c
@@ -20,9 +20,7 @@
 	/* Reference from DFT team: IST_WAIT(10) = 1000 ns  or 1 us delay,
 	 * IST_WAIT(uint32_t pclk_cycles) = usleep(pclk_cycles * PCLK_PERIOD_IN_USEC);
 	 */
-
-	//(pclk % 10) ? udelay((pclk / 10) + 1) : udelay(pclk / 10);
-	udelay((pclk / 1) + 20 + 10000);
+	udelay((pclk / 10) + 10);
 }
 
 int hl_ppboot_sram_self_repair_mismatch_logic(u32 ist_addr, u32 *data,
@@ -57,13 +55,11 @@
 			}
 
 			regmap_writel(data[i], ist_addr);
-			udelay(20);
 			++i;
 
 			if (op == IST_OPC_WRITE_X_COMPARE) {
 				++read_id;
 				rdval = regmap_readl(ist_addr) & data[i];
-				udelay(20);
 				++i;
 				expval = data[i];
 				++i;
@@ -85,7 +81,6 @@
 			} else if (op == IST_OPC_WRITE_BIN_COMPARE) {
 				++read_id;
 				rdval = regmap_readl(ist_addr);
-				udelay(20);
 				expval = data[i];
 				++i;
 
